### PR TITLE
[NPM-3099] Remove source port guessing

### PR DESCRIPTION
**Pull request by hmahmood**
### What does this PR do?

Removes guessing the offset for the source port since this fails fairly regularly in our tests. We were guessing the offset for [`inet_sock->inet_sport`](https://elixir.bootlin.com/linux/v6.6.8/source/include/net/inet_sock.h#L224), which is not necessary since the source port is actually also available from [`sock->skc_num`](https://elixir.bootlin.com/linux/v6.6.8/source/include/net/sock.h#L180), and `inet_sport` is always set from the former.

To determine this is safe to remove:
1. I traced all lines in the kernel source where `inet_sport` gets set. Either `inet_sport` gets set from `skc_num` or `skc_num` gets set to the same value as `inet_sport` together. All these lines have been unchanged since at least 2009.
2. Our offset guess test passes in CI, which tests a wide range of kernels

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by akarpz — APPROVED** [review]

File: pkg/network/tracer/offsetguess/tracer.go [hunk 1]
```diff
@@ -583,7 +569,7 @@ func (t *tracerOffsetGuesser) checkAndUpdateCurrentOffset(mp *ebpf.Map, expected
 		t.status.Offset_rtt, _ = skipOverlaps(t.status.Offset_rtt, t.sockRanges())
 		t.status.Offset_rtt_var = t.status.Offset_rtt + 4
 	case GuessSocketSK:
-		if t.status.Sport_via_sk == htons(expected.sport) && t.status.Dport_via_sk == htons(expected.dport) {
```
Comment (akarpz) [inline pkg/network/tracer/offsetguess/tracer.go:586]: why don't we need this anymore?

Comment (hmahmood) [reply pkg/network/tracer/offsetguess/tracer.go:586]: The `skc_num` field is in host order, not in network order like the `inet_sport` field. See https://elixir.bootlin.com/linux/latest/source/include/net/sock.h#L180 and https://elixir.bootlin.com/linux/latest/source/include/net/inet_sock.h#L224; the latter is declared as `__be16`, which is an annotation to specify the value stored there is going to big endian -- network order.

**Pull request merged**